src/test/java/seedu/rc4hdb/storage/csv/CsvReaderTest.java

```diff
@@ -0,0 +1,161 @@
+package seedu.rc4hdb.storage.csv;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.rc4hdb.testutil.Assert.assertThrows;
+import static seedu.rc4hdb.testutil.TypicalResidents.ALICE;
+import static seedu.rc4hdb.testutil.TypicalResidents.BENSON;
+import static seedu.rc4hdb.testutil.TypicalResidents.CARL;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import seedu.rc4hdb.commons.exceptions.DataConversionException;
+import seedu.rc4hdb.model.ReadOnlyResidentBook;
+import seedu.rc4hdb.model.ResidentBook;
+import seedu.rc4hdb.model.resident.Resident;
+import seedu.rc4hdb.model.resident.fields.Email;
+import seedu.rc4hdb.model.resident.fields.Gender;
+import seedu.rc4hdb.model.resident.fields.House;
+import seedu.rc4hdb.model.resident.fields.MatricNumber;
+import seedu.rc4hdb.model.resident.fields.Name;
+import seedu.rc4hdb.model.resident.fields.Phone;
+import seedu.rc4hdb.model.resident.fields.Room;
+import seedu.rc4hdb.model.resident.fields.Tag;
+import seedu.rc4hdb.storage.csv.exceptions.InvalidCsvFileFormatException;
+import seedu.rc4hdb.testutil.ResidentBuilder;
+
+/**
+ * Unit tests for {@link CsvReader}.
+ */
+public class CsvReaderTest {
+
+    private static final Path TEST_DATA_FOLDER = Paths.get("src", "test", "data", "CsvReaderTest");
+
+    @TempDir
+    public Path testFolder;
+
+    private final CsvReader csvReader = new CsvReader();
+
+    @Test
+    public void readCsvFile_csvFileDoesNotExist_returnEmptyOptional() throws Exception {
```

Comment:
Great work on the testing here. All the tests were adequate but not excessive for the feature so good job on that.